src/catchup/CatchupWork.cpp

```diff
@@ -204,6 +219,11 @@ CatchupWork::runCatchupStep()
                       1;
         mGetHistoryArchiveStateWork =
             addWork<GetHistoryArchiveStateWork>(toCheckpoint, mArchive);
+
+        mGetCurrentStatus = []() -> std::string {
```

Comment:
Is there a particular reason to do it this way instead of implementing `getStatus` in child classes and calling them in `CatchupWork::getStatus`? We're using some of these classes in publishing, and in general, if we were to use them in the future, implementing `getStatus` would prevent the "too vague status logs" issue in the first place. 

It would also be great to add a bit more detail to what we're exposing. Specifically, 
* which ledger am i getting history archive state for?
* which buffered ledger have i applied last? 
* which bucket level am i applying? (currently not implemented)

I'd recommend assessing children works of CatchupWork, and making sure `getStatus` is implemented and either providing some brief but informative message, or calling `getStatus` on its children (when it makes sense, for example in WorkSequence). Then `CatchupWork::getStatus` can call these methods based on whichever work is running.

Reply:
The reasoning was so we wouldn't have to maintain the ordering logic of the jobs in two places. There aren't many jobs though so I think it's actually fine. I removed the `mGetCurrentStatus` mechanism and implemented the missing `getStatus` calls.